best-practices/workflow.md

```diff
@@ -0,0 +1,40 @@
+# Abtion's development workflow
+
+Abtion’s development workflow may vary depending on the project. Teams are encouraged to experiment with the most efficient way to reliably deliver value to our clients. The standard flow is described here. It should be the base from where teams start and the destination, when a team discovers a more efficient way of working.
+
+## Our philosophy:
+
+- We trust our colleges as professional developers.
+- We engage in discussions with an eye on productivity. We accept that a discussion might be counter-productive if the time we spend debating surpasses the benefits of the discussion.
+- We accept that we might be wrong.
+- We want to improve any codebase that we work on.
+- We want to learn.
+- We want to educate our coworkers.
+
+## Goals
+
+- Delivery quality code at a fast pace.
+- Maintain all team members informed of the changes.
+- Have a low defect rate.
+- Have a fast acceptance rate.
+- Be flexible to any sudden change in priority.
+
+## Workflow
+We follow trunk base development. If you are not familiar with it, you can read on https://trunkbaseddevelopment.com. Specifically, we want to avoid large branches/differences between environments. They have been identified as a way to create a knowledge gap in the team and between the client.
+
+- Developers work in pairs.
+- A pair always takes the first card from the backlog. The backlog is already prioritized.
+- A pair will work on the card until is ready. Pair switches might occur during the process.
+- We always branch out from main* and we merge to main.
+- We use short-lived branches to play around with the code.
+- When merging to main, we automatically deploy to production. We may use a test environment** which must always be synchronized with production.
```

Comment:
```suggestion
- When merging to main, we automatically deploy to production if appropriate for the client. We may use a test environment** which must always be synchronized with production with private information obfuscated for security.
```

Reply:
If it is not appropriate, we should use feature flags

Reply:
Agree. In this case I was thinking about far-out examples of firmware or medical software that require a different cycle.